File: frontend/Models/Pedido.cs
using System.ComponentModel.DataAnnotations;

namespace frontendnet.Models;

public class Pedido
{
    [Display(Name = "Id")]
    public int? PedidoId { get; set; }

    [Display(Name = "Usuario")]
    public string? UsuarioEmail { get; set; }

    [Display(Name = "Total")]
    [DataType(DataType.Currency)]
    [DisplayFormat(DataFormatString = "{0:C}")]
    public decimal Total { get; set; }

    [Display(Name = "Estado")]
    public string Estado { get; set; } = "pendiente";

    [Display(Name = "Fecha")]
    [DataType(DataType.DateTime)]
    public DateTime? CreatedAt { get; set; }

    public List<PedidoItem>? Items { get; set; }
}

File: backend/models/pedido.js
'use strict';
const { Model } = require('sequelize');

module.exports = (sequelize, DataTypes) => {
    class pedido extends Model {
        static associate(models) {
            pedido.hasMany(models.pedido_item, { foreignKey: 'pedidoid', as: 'items' });
        }
    }

    pedido.init({
        id: {
            type: DataTypes.INTEGER,
            autoIncrement: true,
            primaryKey: true
        },
        usuarioemail: {
            type: DataTypes.STRING,
            allowNull: false
        },
        total: {
            type: DataTypes.DECIMAL(10, 2),
            allowNull: false
        },
        estado: {
            type: DataTypes.STRING,
            allowNull: false,
            defaultValue: 'pendiente'
        }
    }, {
        sequelize,
        freezeTableName: true,
        modelName: 'pedido',
    });

    return pedido;
};
